### PR TITLE
fix: circular reference issue resolved

### PR DESCRIPTION
**Pull request by SB-rohitdesai**
Mock server not working In case of circular references

**Summary**

Getting 500 Error or Sometimes might need to wait a long time before getting this error when spec having circular references
**Checklist**

- The basics
  - [X] I tested these changes manually in my local or dev environment
- Tests
  - [ ] Added or updated
  - [ ] N/A
- Event Tracking
  - [ ] I added event tracking and followed the event tracking guidelines
  - [ ] N/A
- Error Reporting
  - [ ] I reported errors and followed the error reporting guidelines
  - [ ] N/A

### CODE REVIEW

**Review by bhaskarsontakke — APPROVED** [review]
looks good to me.

**Review by brendarearden — CHANGES_REQUESTED** [review]
could you add a test to cover the 500 scenario?

**Comment by brendarearden** [comment]
is this supposed to include a version of json-schema-ref-parser that has your [optimization changes](https://github.com/stoplightio/json-schema-ref-parser/pull/62) in it? https://github.com/stoplightio/json-schema-ref-parser/pull/62 in it?

**Comment by SB-rohitdesai** [comment]
> could you add a test to cover the 500 scenario?

Test case added

**Comment by SB-rohitdesai** [comment]
> is this supposed to include a version of json-schema-ref-parser that has your [optimization changes](https://github.com/stoplightio/json-schema-ref-parser/pull/62) in it? [stoplightio/json-schema-ref-parser#62](https://github.com/stoplightio/json-schema-ref-parser/pull/62) in it?

Not required. currently we just merging this PR only

**Review by bhaskarsontakke — APPROVED** [review]

**Review by bhaskarsontakke — APPROVED** [review]

File: packages/http/src/mocker/__tests__/HttpMocker.spec.ts [hunk 1]
```diff
@@ -676,5 +689,31 @@ describe('mocker', () => {
         });
       });
     });
+
+    describe('should return 500 error', () => {
```
Comment (brendarearden) [inline packages/http/src/mocker/__tests__/HttpMocker.spec.ts:693]: This test does not seem relevant to the PR that has been opened. Could you please add a test that covers the original circular reference problem and show that it is no longer throwing a 500?

**Review by brendarearden — CHANGES_REQUESTED** [review]

File: packages/http/src/mocker/__tests__/HttpMocker.spec.ts [hunk 2]
```diff
@@ -676,5 +689,142 @@ describe('mocker', () => {
...
+        assertRight(mockResult, result => {
+          expect(result).toMatchObject({
+            statusCode: 500,
+            body: { message: 'Internal Server Error' },
+          });
+        });
+      });
+    });
+
+    describe('With __bundled__ should get response', () => {
```
Comment (brendarearden) [inline packages/http/src/mocker/__tests__/HttpMocker.spec.ts:719]: This test is passing with and without your changes so it still doesn't seem to be testing the initial bug you were seeing / the fix hasn't changed anything. It should be removed or changed to properly capture the issue you are seeing and fixing.

We have [test harness tests](https://github.com/stoplightio/prism/blob/master/test-harness/specs/references/E2E-Recursive-Object-Handling.oas3.txt) for circular references and those have been passing in the pipeline so it seems like circular references are getting resolved in most cases. Maybe a test harness test for the specific scenario that is causing you not to see circular references get resolved would help demonstrate the issue a bit better.

**Pull request closed without merging**